packages/editor/src/editor.ts

```diff
@@ -0,0 +1,236 @@
+    this._loadExternalCode = loadExternalCode;
+    this._editorSession = { cmd: '', args: [], content: '', code: '', tmpDoc: '' };
+    this.print = instanceState.context.print;
+
+    // Add edit command support to shell api.
+    const wrapperFn = (...args: string[]) => {
+      return Object.assign(this.runEditCommand(args), {
+        [Symbol.for('@@mongosh.syntheticPromise')]: true
+      });
+    };
+
+    wrapperFn.isDirectShellCommand = true;
+    wrapperFn.returnsPromise = true;
+    (instanceState.shellApi as any).edit = instanceState.context.edit = wrapperFn;
+    (signatures.ShellApi.attributes as any).edit = {
+      type: 'function',
+      returnsPromise: true,
+      isDirectShellCommand: true
+    } as TypeSignature;
+  }
+
+  activate(): void {
+    this._instanceState.registerPlugin(this);
+  }
+
+  // In the case of using VSCode as an external editor,
+  // we should detect whether the MongoDB extension is installed and open a .mongodb file.
+  // In all other cases we should open a .js file.
+  async _getExtension(cmd: string): Promise<string> {
+    if (!this._isVscodeApp(cmd)) {
+      return 'js';
+    }
+
+    try {
+      const extensions = await fse.readdir(path.join(this._vscodeDir, 'extensions'));
+      const hasMongodbExtension = !!extensions
+        .find((name) => name.includes('mongodb.mongodb-vscode'));
+
+      this.messageBus.emit('mongosh-editor:read-vscode-extensions-done', {
+        vscodeDir: this._vscodeDir,
+        hasMongodbExtension
+      });
+
+      return hasMongodbExtension ? 'mongodb' : 'js';
+    } catch (error) {
+      this.messageBus.emit('mongosh-editor:read-vscode-extensions-failed', {
+        action: 'mongosh-editor:read-vscode-extensions-failed',
+        error: error.message
+      });
+
+      return 'js';
+    }
+  }
+
+  async _getEditor(): Promise<string|null> {
+    // Check for an external editor in the mongosh configuration.
+    let editor: string | null = await this._instanceState.shellApi.config.get('editor');
+
+    // Check for an external editor in the environment variable.
+    if (!editor && process.env.EDITOR) {
+      editor = process.env.EDITOR;
+    }
+
+    return editor;
+  }
+
+  async _createTempFile(): Promise<void> {
+    const ext = await this._getExtension(this._editorSession.cmd);
+    const content = await this._getEditorContent();
+
+    this._editorSession.tmpDoc = path.join(this._tmpDir, `edit-${uuidv4()}.${ext}`);
+
+    // Create a temp file to store content that is being edited.
+    await fse.ensureFile(this._editorSession.tmpDoc);
+    await fse.writeFile(this._editorSession.tmpDoc, beautify(content));
+  }
+
+  async _readTempFile(tmpDoc: string): Promise<string> {
+    const content = await fse.readFile(tmpDoc, 'utf8');
+    await fse.unlink(tmpDoc);
+    return this._makeMultilineJSIntoSingleLine(content);
+  }
+
+  _isVscodeApp(cmd: string): boolean {
+    const regex = /^((.*[\/\\])*)?[cC]ode(.exe)?$/;
```

Comment:
fwiw, this variant of the regexp reduces to:

```suggestion
    const regex = /^(.*)[\/\\]?[cC]ode(.exe)?$/;
```